### PR TITLE
RUST-1012 remove `Clone` impl from `ClientSession`

### PR DESCRIPTION
**Pull request by abr-egn**
RUST-1012

This was included for testing, i.e. so that `Entity` could continue to impl `Clone`; however, cloning a session is somewhere between meaningless and actively harmful.  Updated the few places where `Entity` was actually cloned to use one-off projections for the actual subtypes needed.

### CODE REVIEW

File: src/test/spec/unified_runner/operation.rs [hunk 1]
```diff
@@ -566,10 +568,19 @@ impl TestOperation for Aggregate {
         async move {
             let result = match &self.session {
                 Some(session_id) => {
-                    let entity = test_runner.entities.get(id).unwrap().clone();
+                    enum AggregateEntity {
```
Comment (patrickfreed) [inline src/test/spec/unified_runner/operation.rs:571]: Are these changes necessary? It looks functionally pretty similar to the previous implementation.

Comment (abr-egn) [reply src/test/spec/unified_runner/operation.rs:571]: Yup, it's needed - pre-change, the entity retrieved was unconditionally cloned; with this change, only a collection or database is cloned, other entities are formatted for later debug output.

File: src/test/spec/unified_runner/operation.rs [hunk 2]
```diff
@@ -566,10 +568,19 @@ impl TestOperation for Aggregate {
...
+                        Collection(Collection<Document>),
+                        Database(Database),
+                        Other(String),
+                    }
+                    let entity = match test_runner.entities.get(id).unwrap() {
```
Comment (isabelatkinson) [inline src/test/spec/unified_runner/operation.rs:576]: Instead of going through this helper struct can we match on `test_runner.entities.get(id).unwrap()` and clone within the branches? i.e.
```
match test_runner.entities.get(id).unwrap() {
	Entity::Collection(c) => {
		let c = c.clone();
		...
	}
	...
}
```

**Review by patrickfreed — COMMENTED** [review]
Given that this has API implications, could you file a ticket for it? Then I can add it to the release notes.

File: src/test/spec/unified_runner/operation.rs (hunk 2 above)
Comment (abr-egn) [reply src/test/spec/unified_runner/operation.rs:576]: Not easily, unfortunately - `test_runner.entities.get(id).unwrap()` borrows from `test_runner.entities`, so matching on it would do so for the span of the match, and that would conflict with the exclusive borrow needed for `let session = test_runner.get_mut_session(session_id);`.

**Comment by abr-egn** [comment]
> Given that this has API implications, could you file a ticket for it? Then I can add it to the release notes.

Good call, filed RUST-1012 and updated the PR title/description.

File: src/test/spec/unified_runner/operation.rs (hunk 2 above)
Comment (isabelatkinson) [reply src/test/spec/unified_runner/operation.rs:576]: Ah I do remember that being an issue when I implemented this, I think that's why I had that line of code on 569 in the first place.

**Review by isabelatkinson — APPROVED** [review]

**Review by patrickfreed — APPROVED** [review]
